yeat/workflow/__init__.py

```diff
@@ -49,14 +49,19 @@ def run_workflow(args):
         raise RuntimeError("Snakemake Failed")  # pragma: no cover
 
 
-def resolve_paths(infile):
+def get_config_data(infile):
     data = json.load(open(infile))
     for sample in data["samples"].values():
-        for key, value in sample.items():
-            sample[key] = value if key in DOWNSAMPLE_KEYS else get_resolved_paths(value)
+        resolve_sample_paths(sample)
     return data
 
 
+def resolve_sample_paths(sample):
+    readtypes = set(sample.keys()).intersection(set(READ_TYPES))
+    for readtype in readtypes:
+        sample[readtype] = get_resolved_paths(sample[readtype])
+
+
```

Comment:
Thoughts on the newly refactored function? @standage Other than that, this is PR is ready!